### DISCUSSION TITLE
Change format of request to "subscribe to model change"

### DISCUSSION THREAD

**Discussion by jakiro2017** [NilCoalescing/djangochannelsrestframework#189] [Ideas]
**Is your feature request related to a problem? Please describe.**
I'm feel a bit frustrated when I need to implement model observer subscribe. I want the client send this:
```
{
  "action": "sub",
  "request_id": 42,
  "model": "User"
}
```
instead of 
```
async def websocket_connect(self, message):

    # Super Save
    await super().websocket_connect(message)

    # Initialized operation
    await self.activities_change.subscribe()
```
**Describe the solution you'd like**
Um, same as above?

**Describe alternatives you've considered**
I see https://github.com/theY4Kman/dcrf-client but I want a more simple client from web browser

**Additional context**
Please guide me, I will implement and make pr. Currently, I had a dirty hack by manually add to `available_actions`, but don't know how to call `subscribe()` for model update

**Comment by hishnash** [comment]
So as I understand it you would like to create an `action` were you specify the model type you would like to subscribe it? 

Rather than explicitly setting the model type in the consume you would like the client to be able to select this?

unfortunately due to how django works if we want to be able to observe updates properly we need to have the `@model_observer(..)` statically defined in code.

However if you are ok with adding multiple `@model_observer(...)` declarations to a consumer then you could write an action that using the `model` argument to select the correct model.  

### ManyModelObserveringConsumer

I could also see a mixin (or baseclass) were you might provide at list of models. This could then automatically provide `subscribe` and `unsubscribe` actions. 


```
class MyConsume(ManyModelObserveringConsumer):
    models = [User, Classroom]
```

But this would remove the ability to customise the `groups_for_signal` and `groups_for_consumer` filtering along with the customisation of the message that is sent back to the client so would in many cases result in a large perfomance hit, and users possibly getting notifications about model instances they should not be seeing changes.


### ObserverModelListMixin

For this reason i think it might be better to create a `ObserverModelListMixin` that could be applied to a `GenericAsyncAPIConsumer` in the same way as the `ObserverModelInstanceMixin`. But provide a few extra methods for filtering the subscription.

Then to expose multiple model types I would suggest using a Multiplexer.

**Comment by keezysilencer** [comment]
Regarding a browser alternative to python dcrf-client try [Smart Websocket Client](https://chrome.google.com/webstore/detail/smart-websocket-client/omalebghpgejjiaoknljcfmglgbpocdp/related?utm_source=chrome-app-launcher-info-dialog)
